app/code/ZP/LoyaltyProgram/Model/LoyaltyProgram.php

```diff
@@ -169,4 +176,71 @@ public function setUpdatedAt(?string $updatedAt = null): LoyaltyProgramInterface
     {
         return $this->setData(self::UPDATED_AT, $updatedAt);
     }
+
+    /**
+     * @return int|null
+     */
+    public function getWebsiteId(): ?int
+    {
+        $websiteId = $this->getData(self::WEBSITE_ID);
+
+        return $websiteId === null ? null : (int)$websiteId;
+    }
+
+    /**
+     * @param int|null $websiteId
+     * @return LoyaltyProgramInterface
+     */
+    public function setWebsiteId(?int $websiteId): LoyaltyProgramInterface
+    {
+        return $this->setData(self::WEBSITE_ID, $websiteId);
+    }
+
+    /**
+     * @return array|null
+     */
+    public function getCustomerGroupIds(): ?array
+    {
+        $groupIds = $this->getData(self::CUSTOMER_GROUP_IDS);
```

Comment:
optional: you can specify only array return type . 

`$groupIds = (array)$this->getData(self::CUSTOMER_GROUP_IDS);`

this case null will be converted to empty array